File: Final Project/Food_Ordering_Application/Food_Ordering_Application/Model/Delivery.cs
namespace Food_Ordering_Application.Model
{
    public class Delivery
    {
        public int DeliveryId { get; set; }
        public int OrderId { get; set; } // Foreign key to Order
        public int DeliveryStaffId { get; set; } // Foreign key to User (DeliveryStaff)
        public DateTime PickupTime { get; set; }
        public DateTime? DeliveryTime { get; set; } // Nullable until delivered
        public string? Status { get; set; } // Assigned, PickedUp, Delivered

        public Order? Order { get; set; }
        public User? DeliveryStaff { get; set; }
    }
}

File: Final Project/Food Order Application/Angular/src/app/Delivery.ts
export interface Delivery {
    deliveryId: number;        // Unique ID for the delivery
    orderId: number;           // Foreign key to Order
    deliveryStaffId: number;   // Foreign key to User (Delivery Staff)
    pickupTime?: Date | null;          // Time when the delivery was picked up
    deliveryTime?: Date | null; // Time when the delivery was completed (nullable)
    status: string;            // Current status (Assigned, PickedUp, Delivered)
  }
  